packages/core/src/RenderPipeline/BasicRenderPipeline.ts

```diff
@@ -261,6 +268,26 @@ export class BasicRenderPipeline {
       context.setRenderTarget(colorTarget, colorViewport, mipLevel, cubeFace);
     }
 
+    // Screen space ambient occlusion pass
+    // Before opaque pass so materials can sample ambient occlusion in BRDF
+    const ssaoPass = this._ssaoPass;
+    camera.shaderData.setTexture(Camera._cameraSSAOTextureProperty, engine._basicResources.whiteTexture2D);
+    // Disable macro so PBR won't try to sample unless SSAO is active
+    camera.shaderData.disableMacro(ScreenSpaceAmbientOcclusion._enableMacro);
+    if (ssaoPass) {
+      const depthTexture =
+        camera.depthTextureMode === DepthTextureMode.PrePass
+          ? camera.shaderData.getTexture(Camera._cameraDepthTextureProperty)
+          : null;
+      if (camera.ssao?.enabled && depthTexture && depthTexture !== engine._basicResources.whiteTexture2D) {
+        ssaoPass.onConfig(camera, colorTarget);
+        ssaoPass.onRender(context);
+        // Enable macro globally for PBR
+        camera.shaderData.enableMacro(ScreenSpaceAmbientOcclusion._enableMacro);
+        context.setRenderTarget(colorTarget, colorViewport, mipLevel, cubeFace);
```

Comment:
should be  `context.setRenderTarget(renderTarget, PipelineUtils.defaultViewport, 0);`
do you have test area viewport ?